pkg/controller/backup_sync_controller.go

```diff
@@ -107,4 +125,38 @@ func (c *backupSyncController) run() {
 			}
 		}
 	}
+
+	err = deleteUnused(c, cloudBackups)
+	if err != nil {
+		c.logger.WithError(errors.WithStack(err)).Error("Error deleting unused backup from Kubernetes")
+	}
+}
+
+// deleteUnused deletes backups from Kubernetes if they do not exist in object storage
+func deleteUnused(c *backupSyncController, cloudBackups map[string]string) error {
```

Comment:
Thinking a little more about this, I'd probably change this func to take a `logrus.FieldLogger` (pass `c.logger`), not return an error, and just log immediately if any errors are encountered.  `run()` is just going to log them anyway.